Assignment/CashTest.c

```diff
@@ -0,0 +1,41 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Vinitha Perumal - GET
+// ------------------------------------------------------------------
+// CashTest.c
+// Program on Test2.2 branch.
+// ------------------------------------------------------------------------------------------------
+#include "Cash.h"
+#include <stdio.h>
+
+void TestCases ();
+
+void TestCases () {
+   int cashPaid[] = { 23, 56, 90 }, actualAmount[] = { 22, 37, 40 }, denominations[] = { 10,5,2,1 },
+      expectedOutput[][4] = { { 0, 0, 0, 1 }, { 1, 1, 2, 0 }, { 5, 0, 0, 0 } },
+      numOfTestCases = sizeof (cashPaid) / sizeof (cashPaid[0]), allTestsPassed = 1;
+   for (int i = 0; i < numOfTestCases; i++) {
+      int change = cashPaid[i] - actualAmount[i], coins[4] = { 0 }, isTestPassed = 1,
+         res = Cash (change, denominations, sizeof (denominations) / sizeof (denominations[i]), coins);
+      for (int j = 0; j < 4 && res == 1; j++) {
+         if (coins[j] != expectedOutput[i][j]) {
+            isTestPassed = 0;

```

Comment:
instead of assigning isTestPassed = 0, simply print "Test failed" with appropriate info and exit function.